### PR TITLE
statstics: fix update the value of LFU's MaxCost

### LINKED ISSUE #46585: improve memory usage for stats LFU cache

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #46585

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 1]
```diff
@@ -41,26 +41,33 @@ type LFU struct {
 
 var testMode = false
 
-// NewLFU creates a new LFU cache.
-func NewLFU(totalMemCost int64) (*LFU, error) {
+func setMemCost(totalMemCost int64) (result int64, err error) {
```
Comment (qw4990) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:44]: Better to rename this function to `adjustMemCost` and move `if intest.InTest {...}` in to this function.

Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go:44]: In fact, when to set the capacity, we don‘t use this ```intest``` logic.

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 2]
```diff
@@ -41,26 +41,33 @@ type LFU struct {
 
 var testMode = false
 
-// NewLFU creates a new LFU cache.
-func NewLFU(totalMemCost int64) (*LFU, error) {
+func adjustMemCost(totalMemCost int64) (result int64, err error) {
```
Comment (hi-rustin) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:44]: Could you please add some comments about this logic?

**Review by hi-rustin — COMMENTED** [review]
Thanks!

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 3]
```diff
@@ -41,26 +41,33 @@ type LFU struct {
...
+}
+
+// NewLFU creates a new LFU cache.
+func NewLFU(totalMemCost int64) (*LFU, error) {
+	cost, err := adjustMemCost(totalMemCost)
+	if err != nil {
+		return nil, err
+	}
+	if intest.InTest {
+		cost = 5000000
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply statistics/handle/cache/internal/lfu/lfu_cache.go:65]: Done

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**